extensions/amp-date-countdown/1.0/date-countdown.js

```diff
@@ -0,0 +1,247 @@
+    ...localeStrings,
+  };
+  return (
+    <div ref={rootRef} {...rest}>
+      {render(data, children)}
+    </div>
+  );
+}
+
+/**
+ * Calculate the epoch time that this component should countdown to from
+ * one of multiple input options.
+ * @param {string|undefined} endDate
+ * @param {number|undefined} timeleftMs
+ * @param {number|undefined} timestampMs
+ * @param {number|undefined} timestampSeconds
+ * @return {number}
+ */
+function getEpoch(endDate, timeleftMs, timestampMs, timestampSeconds) {
+  let epoch;
+
+  if (endDate) {
+    epoch = Date.parse(endDate);
+  } else if (timeleftMs) {
+    epoch = Date.now() + timeleftMs;
+  } else if (timestampMs) {
+    epoch = timestampMs;
+  } else if (timestampSeconds) {
+    epoch = timestampSeconds * 1000;
+  }
+
+  if (epoch === undefined) {
+    throw new Error(
+      `One of endDate, timeleftMs, timestampMs, timestampSeconds` +
+        `is required. ${NAME}`
+    );
+  }
+  return epoch;
+}
+
+/**
+ * Return an object with a label for 'years', 'months', etc. based on the
+ * user provided locale string.
+ * @param {string} locale
+ * @return {!JsonObject}
+ */
+function getLocaleWord(locale) {
+  if (getLocaleStrings(locale) === undefined) {
+    displayWarning(
+      `Invalid locale ${locale}, defaulting to ${DEFAULT_LOCALE}. ${NAME}`
+    );
+    locale = DEFAULT_LOCALE;
+  }
+  const localeWordList = getLocaleStrings(locale);
+  return {
+    'years': localeWordList[0],
+    'months': localeWordList[1],
+    'days': localeWordList[2],
+    'hours': localeWordList[3],
+    'minutes': localeWordList[4],
+    'seconds': localeWordList[5],
+  };
+}
+
+/**
+ * Converts a time represented in milliseconds (ms) into a representation with
+ * days, hours, minutes, etc. and returns formatted strings in an object.
+ * @param {number} ms
+ * @param {string} biggestUnit
+ * @return {JsonObject}
+ */
+function getYDHMSFromMs(ms, biggestUnit) {
+  //Math.trunc is used instead of Math.floor to support negative past date
+  const d =
+    TimeUnit[biggestUnit] == TimeUnit.DAYS
+      ? supportBackDate(Math.floor(ms / MILLISECONDS_IN_DAY))
+      : 0;
+  const h =
+    TimeUnit[biggestUnit] == TimeUnit.HOURS
+      ? supportBackDate(Math.floor(ms / MILLISECONDS_IN_HOUR))
+      : TimeUnit[biggestUnit] < TimeUnit.HOURS
+      ? supportBackDate(
+          Math.floor((ms % MILLISECONDS_IN_DAY) / MILLISECONDS_IN_HOUR)
+        )
+      : 0;
+  const m =
+    TimeUnit[biggestUnit] == TimeUnit.MINUTES
+      ? supportBackDate(Math.floor(ms / MILLISECONDS_IN_MINUTE))
+      : TimeUnit[biggestUnit] < TimeUnit.MINUTES
+      ? supportBackDate(
+          Math.floor((ms % MILLISECONDS_IN_HOUR) / MILLISECONDS_IN_MINUTE)
+        )
+      : 0;
+  const s =
+    TimeUnit[biggestUnit] == TimeUnit.SECONDS
+      ? supportBackDate(Math.floor(ms / MILLISECONDS_IN_SECOND))
+      : supportBackDate(
+          Math.floor((ms % MILLISECONDS_IN_MINUTE) / MILLISECONDS_IN_SECOND)
+        );
+
+  return {
+    d,
```

Comment:
I believe these still should be in quotes. /cc @jridgewell

Reply:
Updated, thanks!